lib/mlir/Dialect/Clift/Utils/CBackend.cpp

```diff
@@ -0,0 +1,1247 @@
+    llvm::SmallString<64> String;
+    {
+      llvm::raw_svector_ostream Stream(String);
+
+      if (Signed and static_cast<int64_t>(Value) < 0) {
+        Stream << static_cast<int64_t>(Value);
+      } else {
+        Stream << Value;
+      }
+
+      Stream << getCIntegerLiteralSuffix(Integer, Signed);
+    }
+    return C.getConstantTag(String);
+  }
+
+  ptml::Tag getIntegerConstant(uint64_t Value, PrimitiveType Type) {
+    auto [Integer, Signed] = getCInteger(Platform, Type);
+    return getIntegerConstant(Value, Integer, Signed);
+  }
+
+  void emitIntegerImmediate(uint64_t Value, ValueType Type) {
+    Type = dealias(Type, /*IgnoreQualifiers=*/true);
+
+    if (auto T = mlir::dyn_cast<PrimitiveType>(Type)) {
+      Out << getIntegerConstant(Value, T);
+    } else {
+      auto TypeAttr = mlir::cast<DefinedType>(Type).getElementType();
+      const auto &ModelType = getModelTypeDefinition(TypeAttr);
+      const auto &ModelEnum = llvm::cast<model::EnumDefinition>(ModelType);
+
+      auto It = ModelEnum.Entries().find(Value);
+      if (It == ModelEnum.Entries().end())
+        revng_abort("Model enum entry not found.");
+
+      Out << C.getLocation(/*IsDefinition=*/false, ModelEnum, *It);
+    }
+  }
+
+  //===---------------------------- Expressions ---------------------------===//
+
+  RecursiveCoroutine<void> emitImmediateExpression(mlir::Value V) {
+    auto E = V.getDefiningOp<ImmediateOp>();
+    emitIntegerImmediate(E.getValue(), E.getResult().getType());
+    rc_return;
+  }
+
+  RecursiveCoroutine<void> emitParameterExpression(mlir::Value V) {
+    auto Arg = mlir::cast<mlir::BlockArgument>(V);
+    Out << ParameterNames[Arg.getArgNumber()];
+    rc_return;
+  }
+
+  RecursiveCoroutine<void> emitLocalVariableExpression(mlir::Value V) {
+    // TODO: Emit variable name from the model once the model is extended to
+    //       provide this information.
+
+    auto Local = V.getDefiningOp<LocalVariableOp>();
+    Out << C.getVariableLocationReference(Local.getSymName(), *CurrentFunction);
+
+    rc_return;
+  }
+
+  RecursiveCoroutine<void> emitUseExpression(mlir::Value V) {
+    auto E = V.getDefiningOp<UseOp>();
+
+    auto Module = E->getParentOfType<clift::ModuleOp>();
+    revng_assert(Module);
+
+    mlir::Operation
+      *SymbolOp = mlir::SymbolTable::lookupSymbolIn(Module,
+                                                    E.getSymbolNameAttr());
+    revng_assert(SymbolOp);
+
+    if (auto G = mlir::dyn_cast<GlobalVariableOp>(SymbolOp)) {
+      Out << C.getLocationReference(getModelSegment(G));
+    } else if (auto F = mlir::dyn_cast<FunctionOp>(SymbolOp)) {
+      auto Visitor = [&](const auto *ModelFunction) {
+        Out << C.getLocationReference(*ModelFunction);
+      };
+      std::visit(Visitor, getModelFunctionVariant(F));
+    } else {
+      revng_abort("Unsupported global operation");
+    }
+
+    rc_return;
+  }
+
+  template<typename Aggregate>
```

Comment:
Rename `Aggregate` to `Class` would better match the naming adopted in all the rest.